src/mgr/Mgr.cc

```diff
@@ -219,12 +219,35 @@ std::map<std::string, std::string> Mgr::load_store()
   return loaded;
 }
 
+void Mgr::handle_signal(int signum)
+{
+  ceph_assert(signum == SIGINT || signum == SIGTERM);
+  derr << "*** Got signal " << sig_str(signum) << " ***" << dendl;
+  shutdown();
```

Comment:
can we add a method to `PyModuleRegistry`, so we can also shutdown and reset `standby_modules` as well?

Reply:
There's already there: https://github.com/ceph/ceph/blob/master/src/mgr/PyModuleRegistry.cc#L224

We can invoke that from `Mgr::shutdown()` and probably call that as `standby_shutdown()`?

Reply:
no, we cannot. see https://tracker.ceph.com/issues/38621

Reply:
fair enough -- so, no other changes required to this PR, right?

Reply:
i still suggest https://github.com/ceph/ceph/pull/30890#discussion_r336576088 , but i don't think it's a blocker. please feel free to ignore it, if you think it's not a must. in that case, i'll post a follow-up PR to address it.